src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -0,0 +1,98 @@
+#include "effects/backends/builtin/pitchshifteffect.h"
+
+#include "util/sample.h"
+
+PitchShiftGroupState::~PitchShiftGroupState() noexcept {
+    SampleUtil::free(m_retrieveBuffer[0]);
+    SampleUtil::free(m_retrieveBuffer[1]);
+
+    VERIFY_OR_DEBUG_ASSERT(m_pRubberBand) {
```

Comment:
All this code is probably redundant.

Reply:
I couldn't find the other part of the code, where it should be done. This code was based on how other built-in effects use `SampleUtil::free` in their destructors. For example, see:

https://github.com/mixxxdj/mixxx/blob/5158e6259cfbba2f71e4f112d642259c48659354/src/effects/backends/builtin/linkwitzriley8eqeffect.cpp#L53

or

https://github.com/mixxxdj/mixxx/blob/5158e6259cfbba2f71e4f112d642259c48659354/src/effects/backends/builtin/lvmixeqbase.h#L47

Reply:
This can be a normal debug assert.